src/lotus/message/ipc.rs

```diff
@@ -115,23 +119,65 @@ pub struct CheckpointData {
     #[serde(rename(deserialize = "PrevCheck"))]
     pub prev_check: Option<CIDMap>,
     #[serde(rename(deserialize = "CrossMsgs"))]
-    pub cross_msgs: Option<CrossMsgMetaWrapper>,
+    pub cross_msgs: Option<BatchCrossMsgWrapper>,
 }
 
 #[derive(PartialEq, Eq, Clone, Debug, Default, Deserialize, Serialize)]
-pub struct CrossMsgMetaWrapper {
-    #[serde(rename(deserialize = "MsgsCid"))]
-    pub msgs_cid: Option<CIDMap>,
-    #[serde(rename(deserialize = "Nonce"))]
-    pub nonce: u64,
-    #[serde(rename(deserialize = "Value"))]
-    #[serde(deserialize_with = "deserialize_token_amount_from_str")]
-    pub value: TokenAmount,
+pub struct BatchCrossMsgWrapper {
+    #[serde(rename(deserialize = "CrossMsgs"))]
+    pub cross_msgs: Option<Vec<CrossMsgsWrapper>>,
     #[serde(rename(deserialize = "Fee"))]
     #[serde(deserialize_with = "deserialize_token_amount_from_str")]
     pub fee: TokenAmount,
 }
 
+impl From<BatchCrossMsgWrapper> for BatchCrossMsgs {
+    fn from(wrapper: BatchCrossMsgWrapper) -> Self {
+        let cross_msgs = wrapper.cross_msgs.map(|cross_msgs| {
+            cross_msgs
+                .into_iter()
+                .map(|cross_wrapper| CrossMsg {
+                    msg: StorableMsg {
+                        from: cross_wrapper.msg.from,
+                        to: cross_wrapper.msg.to,
+                        method: cross_wrapper.msg.method,
+                        params: cross_wrapper.msg.params,
+                        value: cross_wrapper.msg.value,
+                        nonce: cross_wrapper.msg.nonce,
+                    },
+                    wrapped: cross_wrapper.wrapped,
+                })
+                .collect::<Vec<_>>()
+        });
+
+        BatchCrossMsgs {
+            cross_msgs,
+            fee: wrapper.fee,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct CrossMsgsWrapper {
+    pub msg: StorableMsgsWrapper,
+    pub wrapped: bool,
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct StorableMsgsWrapper {
+    // TODO: @will,IPCAddress is currently serialized by default as a tuple,
+    // we need to implement its map counterpart so it can be deserialized
+    // using a map from Lotus.
+    pub from: IPCAddress,
```

Comment:
It would be great if we can de-risk the `serde` of this one from Lotus as soon as possible, this one is also needed by top-down messages. Let's sync tomorrow and test this together if needed.

Reply:
updated, please check